notebooks/algorithm-design/instances.ipynb

```diff
@@ -0,0 +1,284 @@
+    "\\hat{H}_{k-1} := \n",
+    "\\hat{H} + \\sum_{j=0}^{k-2}\\beta_j |\\psi(\\vec{\\theta^j})\\rangle \\langle \\psi(\\vec{\\theta^j})|. \n",
+    "$$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Subspace Search VQE (SSVQE)\n",
+    "\n",
+    "[SSVQE](https://arxiv.org/abs/1810.09434) is a variant of VQE that, like VQD, enables you to obtain the first $k$ eigenvalues of an observable $\\hat{H}$ with eigenvalues $\\{\\lambda_0, \\lambda_1,...,\\lambda_{N-1}\\}$, with $N\\geq k$. We'll assume without loss of generality that $\\lambda_0<\\lambda_1<...<\\lambda_{N-1}$.\n",
+    "\n",
+    "This algorithm requires the usage of $k$ mutually orthogonal reference states ${|\\rho_j\\rangle}_{j=0}^{k-1}$, that is, $\\langle \\rho_j | \\rho_l \\rangle = \\delta_{jl}$ for $j,l<k$. \n",
+    "\n",
+    "The (only) cost function of this algorithm is then:\n",
+    "\n",
+    "$$\n",
+    "C(\\vec{\\theta}) := \\sum_{j=0}^{k-1} w_j \\langle \\rho_j | U^{\\dagger}(\\vec{\\theta})\\hat{H} U(\\vec{\\theta})|\\rho_j \\rangle\n",
+    "$$\n",
+    "\n",
+    "where $w_j$ is an arbitrary positive number such that if $j<l<k$ then $w_j>w_l$ and $U(\\vec{\\theta})$ is the user-defined variational form.\n",
+    "\n",
+    "Like VQD, SSVQE hinges on the idea that eigenstates that correspond to different eigenvalues are mutually orthogonal. In particular, the inner product of $U(\\vec{\\theta})|\\rho_j\\rangle$ and $U(\\vec{\\theta})|\\rho_l\\rangle$ is given by:\n",
+    "\n",
+    "$$\n",
+    "\\langle \\rho_j | U^{\\dagger}(\\vec{\\theta})U(\\vec{\\theta})|\\rho_l \\rangle = \\langle \\rho_j | I |\\rho_l \\rangle = \\langle \\rho_j | \\rho_l \\rangle = \\delta_{jl}\n",
+    "$$\n",
+    "\n",
+    "The first equality holds because the ansatz $U(\\vec{\\theta})$ is a quantum operator and therefore unitary and the last one, because of the orthogonality of the reference states $|\\rho_j\\rangle$. This fact that orthogonality is preserved through unitary transformations is deeply related to the principle of conservation of information, as expressed in quantum information science. Under this view, non-unitary transformations represent processes where information is either lost or injected.\n",
+    "\n",
+    "The question that arises now is: How can we ensure that those states are all eigenstates? That's where the weights $w_j$ become important. The idea is that, if the weights are different enough, the term with largest weight (i.e. $w_0$) will be given priority during optimization over the rest, so the resulting state $U(\\vec{\\theta})|\\rho_0 \\rangle$ will end up becoming the eigenstate corresponding to $\\lambda_0$. Then, because of the mutual orthogonality of ${U(\\vec{\\theta})|\\rho_j\\rangle}_{j=0}^{k-1}$, the rest of the states will be orthogonal to it and, therefore, contained in the subspace corresponding to the eigenvalues $\\{\\lambda_1,...,\\lambda_{N-1}\\}$.\n",
```

Comment:
```suggestion
    "The question that arises now is: How can we ensure that those states are all eigenstates? That's where the weights $w_j$ become important. The idea is that, if the weights are different enough, the term with largest weight (i.e. $w_0$) will be given priority during optimization over the rest, so the resulting state $U(\\vec{\\theta})|\\rho_0 \\rangle$ will end up becoming the eigenstate corresponding to $\\lambda_0$. Then, because of the mutual orthogonality of $U(\\vec{\\theta})|\\rho_j\\rangle_{j=0}^{k-1}$, the rest of the states will be orthogonal to it and, therefore, contained in the subspace corresponding to the eigenvalues $\\{\\lambda_1,...,\\lambda_{N-1}\\}$.\n",
```